hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/OneReplicaPipelineSafeModeRule.java

```diff
@@ -75,69 +66,59 @@ public OneReplicaPipelineSafeModeRule(String ruleName, EventQueue eventQueue,
             HDDS_SCM_SAFEMODE_ONE_NODE_REPORTED_PIPELINE_PCT  +
             " value should be >= 0.0 and <= 1.0");
 
+    // Exclude CLOSED pipeline
     int totalPipelineCount =
         pipelineManager.getPipelines(HddsProtos.ReplicationType.RATIS,
-            HddsProtos.ReplicationFactor.THREE).size();
+            HddsProtos.ReplicationFactor.THREE, Pipeline.PipelineState.OPEN)
+            .size() +
+            pipelineManager.getPipelines(HddsProtos.ReplicationType.RATIS,
+                HddsProtos.ReplicationFactor.THREE,
+                Pipeline.PipelineState.ALLOCATED).size();
```

Comment:
The allocated pipeline does not guarantee to be usable on DNs. Should we exclude them from the calculation here?

Reply:
Hi  @xiaoyuyao , here are two cases, 
1.  A new cluster, ALLOCATED pipelie numbe is 0 at cluster start up. 
2.  A running cluster, restart SCM,  all the pipelines loaded from DB is marked as ALLOCATED(SCMPipelineManager#initializePipelineState(), current logic). So in this case, ALLOCATED state pipeline means OPEN pipeline actually.